src/onboard_flash.rs

```diff
@@ -791,16 +805,213 @@ impl OnboardFlash {
         self.spi_transfer(&mut id);
         info!("Device id {:x}", id);
     }
-    pub fn start_file(&mut self) -> isize {
+    pub fn start_file(&mut self, start_page: isize) -> isize {
         // We always start writing a new file at page 1, reserving page 0 for when we come back
         // and write the header once we've finished writing the file.
-        self.current_page_index = 1;
+        self.current_page_index = start_page;
         warn!(
             "Starting file at file block {}, page {}",
             self.current_block_index, self.current_page_index
         );
         self.current_block_index
     }
+    pub fn finish_transfer(
```

Comment:
Hmm, maybe these async functions could actually work if you fix the issue with the memory being passed to the DMA not actually being valid?